File: mobile/src/main/java/info/papdt/express/helper/receiver/ConnectivityReceiver.java
package info.papdt.express.helper.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;
import android.util.Log;

import info.papdt.express.helper.support.PushUtils;

public class ConnectivityReceiver extends BroadcastReceiver {

	@Override
	public void onReceive(Context context, Intent intent) {
		if (readNetworkState(context)) {
			Log.i("ConnectivityReceiver", "Start ReminderService.");
			PushUtils.startServices(context);
		} else {
			Log.i("ConnectivityReceiver", "Stop ReminderService.");
			PushUtils.stopServices(context);
		}
	}

	public static boolean readNetworkState(Context context) {
		if (context == null) {
			return false;
		}

		ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
		return cm != null && cm.getActiveNetworkInfo() != null && cm.getActiveNetworkInfo().isConnected();
	}

}

File: mobile/src/main/kotlin/info/papdt/express/helper/receiver/ConnectivityReceiver.kt
package info.papdt.express.helper.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import info.papdt.express.helper.support.PushUtils
import moe.feng.kotlinyan.common.*

class ConnectivityReceiver : BroadcastReceiver() {

	override fun onReceive(context: Context, intent: Intent) {
		context.run(if (readNetworkState(context)) PushUtils::startServices else PushUtils::stopServices)
	}

	companion object {

		fun readNetworkState(context: Context?) = context?.connectivityManager?.activeNetworkInfo?.isConnected ?: false

	}

}